hail/src/main/scala/is/hail/expr/ir/InferPType.scala

```diff
@@ -447,13 +447,6 @@ object InferPType {
           theIR._pType2
         }))
       case In(_, pType: PType) => pType
-      case ArrayFor(a, valueName, body) =>
```

Comment:
Why can ArrayFor go away? Is ArrayFor never noSharing (in github search at least I can’t find this property, will check again in IntelliJ when I get off mobile)? I didn’t see a simplify rule that would come before inferptype that would obviate need to infer on ArrayFor, since streamify keeps it and the only simplify rule applies to a Begin body. Want to make sure I understand.

Reply:
it managed to sneak in twice in a rebase mistake (I'm assuming) :)

Reply:
definitely do need an infer rule for arrayfor!